spec/integration/schema_parser_spec.rb

```diff
@@ -60,6 +68,67 @@ def to_dto
           expect(testhandler.to_dto).to eq(OpenStruct.new(body: { name: 'About naming stuff and cache invalidation' }))
         end
       end
+      context 'valid schema with included objects' do
+        let(:invalid_body) do
+          <<-JSON
+          {
+            "data": {
+              "attributes": {
+                "name": "About naming stuff and cache invalidation"
+              }
+            },
+            "included": [
+              {
+                "type": "my_types",
+                "attributes": {
+                  "some_random_number": 1337
+                }
+              }
+            ]
+          }
+        JSON
+        end
+        let(:testclass) do
+          Class.new(RequestHandler::Base) do
+            options do
+              body do
+                schema(Dry::Validation.JSON do
+                         required(:name).filled(:str?)
+                       end)
+
+                included do
+                  my_types(Dry::Validation.JSON do
+                             required(:view_count).filled(:int?)
+                           end)
+                end
+              end
+            end
+            def to_dto
+              OpenStruct.new(
+                body:  body_params
+              )
+            end
+          end
+        end
+        it 'raises a SchemaValidationError with invalid data in included type' do
+          request = build_mock_request(params: {}, headers: {}, body: invalid_body)
+          testhandler = testclass.new(request: request)
+          expect { testhandler.to_dto }.to raise_error(RequestHandler::SchemaValidationError)
+        end
+
+        it 'raises a MissingArgumentError with missing data' do
+          request = instance_double('Rack::Request', params: {}, env: {}, body: nil)
+          testhandler = testclass.new(request: request)
+          expect { testhandler.to_dto }.to raise_error(RequestHandler::MissingArgumentError)
+        end
+
+        it 'works for valid data' do
+          request = build_mock_request(params: {}, headers: {}, body: valid_body)
+          testhandler = testclass.new(request: request)
+          expect(testhandler.to_dto)
+            .to eq(OpenStruct.new(body: [{ name: 'About naming stuff and cache invalidation' }, { view_count: 1337 }]))
+        end
```

Comment:
this is missing the case where the schema defines some included schemas but the body doesn't have any (which is fine as the relationship entities could already be present)